package/MDAnalysis/coordinates/XDR.py

```diff
@@ -197,11 +197,13 @@ def _load_offsets(self):
         try:
             with fasteners.InterProcessLock(lock_name) as filelock:
                 pass
-        except PermissionError:
-            warnings.warn(f"Cannot write lock/offset file in same location as "
-                           "{self.filename}. Using slow offset calculation.")
-            self._read_offsets(store=True)
-            return
+        except OSError as e:
+            if isinstance(e, PermissionError) or e.errno == errno.EROFS:
+                warnings.warn(f"Cannot write lock/offset file in same location as "
+                               "{self.filename}. Using slow offset calculation.")
+                self._read_offsets(store=True)
+                return
```

Comment:
```suggestion
            else:
```

Reply:
I'm assuming you didn't mean to remove the return statement. ?

Reply:
You're absolutely right, that was bogus